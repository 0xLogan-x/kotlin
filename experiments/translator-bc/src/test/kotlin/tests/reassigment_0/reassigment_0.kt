fun reassigment_0(x: Int): Int {
    val b = x + 1
    return b
}