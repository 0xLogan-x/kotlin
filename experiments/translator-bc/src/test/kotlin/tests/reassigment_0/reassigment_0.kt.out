@exceptions.str.1 = private unnamed_addr constant  [59 x i8] c"Exception in thread main kotlin.KotlinNullPointerException\00", align 1
declare void @llvm.memcpy.p0i8.p0i8.i64(i8* nocapture, i8* nocapture readonly, i64, i32, i1)
declare i8* @malloc_heap(i32)
declare i32 @printf(i8*, ...)
declare void @abort()
%class.Nothing = type { }
define weak i32 @reassigment_0_Int(i32  %reassigment_0.x) #0
{
%reassigment_0.x.addr = alloca i32, align 4
store i32 %reassigment_0.x, i32* %reassigment_0.x.addr, align 4
%var2 = load i32* %reassigment_0.x.addr, align 4
%var3 = add nsw i32 %var2, 1
%managed.unique.0.reassigment_0.b = alloca i32, align 4
%var4 = load i32* %managed.unique.0.reassigment_0.b, align 4
store i32 %var3, i32* %managed.unique.0.reassigment_0.b, align 4
%var5 = load i32* %managed.unique.0.reassigment_0.b, align 4
ret i32 %var5
unreachable
}

